modules/tinymce/src/plugins/table/main/ts/core/BorderNormalizer.ts

```diff
@@ -0,0 +1,73 @@
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Arr, Optional } from '@ephox/katamari';
+import { Css, SugarElement } from '@ephox/sugar';
+import { DomModifier } from '../ui/DomModifier';
+
+const getComputedCombinedBorderProperty = (el: SugarElement<HTMLTableCellElement>, property: string) => {
+  const getSideValue = (side: string) => {
+    const propName = `border-${side}-${property}`;
+
+    // IE doesn't return the right computed style value even if it's explicitly set in the style
+    // attribute so we need to first check the raw value since that is the common case anyway
+    return Css.getRaw(el, propName).getOr(Css.get(el, propName));
+  };
+
+  const values = Arr.map([ 'top', 'left', 'bottom', 'right' ], getSideValue);
+  return values[0] !== '' && Arr.forall(values.slice(1), (v) => v === values[0]) ? Optional.some(values[0]) : Optional.none();
+};
+
+// IE returns subpixel border values even if the css is set to 1px so lets normalize that to the closest pixel
+const parsePxValue = (w: string) => Math.round(parseFloat(w));
+
+const getComputedBorderWidth = (el: SugarElement<HTMLTableCellElement>) => getComputedCombinedBorderProperty(el, 'width').map(parsePxValue).getOr(1);
+const getComputedBorderStyle = (el: SugarElement<HTMLTableCellElement>) => getComputedCombinedBorderProperty(el, 'style').getOr('solid');
+
+const setBorderStyle = (modifier: DomModifier, style: string) => modifier.setFormat('tablecellborderstyle', style);
+const setBorderWidth = (modifier: DomModifier, width: string) => modifier.setFormat('tablecellborderwidth', width);
+
+const isVisibleBorderStyle = (value: string) => value !== 'none' && value !== 'hidden' && value !== '';
+const hasVisibleBorderStyle = (el: SugarElement<HTMLTableCellElement>) => isVisibleBorderStyle(getComputedBorderStyle(el));
+const hasBorderColor = (el: SugarElement<HTMLTableCellElement>) => Css.getRaw(el, 'border-color').isSome();
+const hasOnePxBorderWidth = (el: SugarElement<HTMLTableCellElement>) => getComputedBorderWidth(el) === 1;
+
+export const normalizeSetCellBorderColor = (modifier: DomModifier, cell: SugarElement<HTMLTableCellElement>, value: string) => {
+  if (hasOnePxBorderWidth(cell) && hasVisibleBorderStyle(cell)) {
+    // We need to set it to unset the border-style when we are clearing the border-color to prevent the border from overlapping to adjacent cells
+    setBorderStyle(modifier, value === '' ? '' : 'double');
+  }
+};
+
+export const normalizeSetCellBorderWidth = (modifier: DomModifier, cell: SugarElement<HTMLTableCellElement>, value: string) => {
+  const width = value === '' ? 0 : parseFloat(value);
+
+  if (hasVisibleBorderStyle(cell)) {
+    if (width === 1) {
+      // We need to set border-style to `double` in order render the border-color
+      setBorderStyle(modifier, hasBorderColor(cell) ? 'double' : '');
```

Comment:
This also breaks things if the previous border style was not single/double. Eg if it was dotted and then I change the width all of a sudden the style disappears.

Reply:
That is by design yes. All border-styles except solid will not render properly in 1px and our default is solid for tables so it makes sense to me to just clear it then.

Reply:
The editor team were in a call and we found that very confusing. I think this whole concept needs more thought and planning before implementation honestly. When we had the call today, we are planning to have a call with Product and HV to determine how to move forward with this.

Reply:
I should also add that behaviour is not how other editors work (eg google docs or word). They will keep the correct styling when changing the widths.

Reply:
Yes, but they have a dedicated rendering engine not html as their base. So we can't compare apples and oranges here in this case. The logic here might be confusing yes, that is why I tried to back it up with comments, tests and details in the ticket for these different cases.

Reply:
One thing that google docs does though is that it doesn't style the TD:s with borders that makes it possible to apply 1px styles to the cells and render them. However I remember asking Fredrik about why our content skin is setting styles on TD that and got the answer that that is a common way to do that. But maybe we should just alter the css to not do that then we wouldn't need this normalizer at all and instruct people that have their own css to not do that either. I will double check that it works correctly first on all platforms we support.

Reply:
I immediately hit an issue with that google docs renders the table differently in FF vs Chrome. It will put border: 1px solid black on all cells on FF but on Chrome it will fallback to the default border colors of the table since that is how Chrome renders borders. We could in theory do this once we have a separate rendering of the view with the alt model but until then this would not work or produce super verbose html since we then would have to normalize the whole table.